compiler/noirc_evaluator/src/ssa/opt/loop_invariant.rs

```diff
@@ -109,6 +110,28 @@ impl<'f> LoopInvariantContext<'f> {
 
                 if hoist_invariant {
                     self.inserter.push_instruction(instruction_id, pre_header);
+
+                    // We track whether we may mutate MakeArray instructions before we deduplicate
+                    // them but we still need to issue an extra inc_rc in case they're mutated afterward.
```

Comment:
> We track whether we may mutate MakeArray instructions before we deduplicate them

Where do you do this? Looks like this comment was just copy pasted from constant folding

Reply:
Ah yeah, I did not read the comment carefully enough. I have updated it to just say we must issue an inc_rc instruction.